src/main/java/org/jetbrains/bsp/bazel/server/bsp/resolvers/actiongraph/ActionGraphV2Parser.java

```diff
@@ -0,0 +1,109 @@
+package org.jetbrains.bsp.bazel.server.bsp.resolvers.actiongraph;
+
+import com.google.common.collect.Lists;
+import com.google.devtools.build.lib.analysis.AnalysisProtosV2;
+import java.util.*;
+import java.util.stream.Collectors;
+import org.jetbrains.bsp.bazel.commons.Uri;
+
+public class ActionGraphV2Parser implements ActionGraphParser {
+  private final AnalysisProtosV2.ActionGraphContainer actionGraph;
+  private final Map<Integer, AnalysisProtosV2.PathFragment> pathFragmentMap;
+
+  public ActionGraphV2Parser(AnalysisProtosV2.ActionGraphContainer actionGraph) {
```

Comment:
could you add a link to the specification of what we're actually parsing here?

Reply:
also I think I could receive some unit testing love

Reply:
> could you add a link to the specification of what we're actually parsing here?

If you have the protocol buffers plugin it already redirects you to the .proto file of the implementation. Do you still think it makes sense?

Reply:
> also I think I could receive some unit testing love

You mean we should have some unit tests for the action graph parsers? How would we go about doing those? Would we add some example binaries to the resources?

Reply:
I should have expressed it in a better way, but what I mean is not the specification of the proto itself, but rather explanation of what this parser does. Currently it appears to be a black box. As you mentioned in one of your comments, this parser shares a lot of logic with the V1 parser, but it is not clear what this logic is supposed to do.

Reply:
Regarding the unit tests, what I meant was related to the problem with the queue. There certainly was a mistake within the method logic (new queue with a single element being created each time), but we didn't have the tests which would allow us to spot this mistake. The tests could also act as a specification. I would like to know what's the expected output of all these methods - and I could potentially learn that by reading the tests, but there are none

Reply:
I'm not sure what you mean by example binaries though. I imagine that these tests could use some hand-crafted instances of the class `AnalysisProtosV2.ActionGraphContainer`